### DISCUSSION TITLE
about the HopToDesk fork

### DISCUSSION THREAD

**Discussion by zeroheure** [rustdesk/rustdesk#1590] [Q&A]
Hi
I've just tried [HopToDesk](hoptodesk.com) and browsed its source. It is a fork, but is there hostile relation ship between the 2 projects ?

**Comment by zeroheure** [comment]
I've looked a bit in some HopToDesk files (HopToDesk is on Gitlab):
- The owner has removed every mention of Rustdesk in the source. AFAIK, the AGPL3 don't allow to do that. 
- He doesn't know how to setup a repositoy with Git, source is a zip file and he asked for my help.

I will send a polite email requesting him to respect the license.

**Reply by rustdesk** [reply]
Thanks

**Reply by zeroheure** [reply]
Here's a copy of my email to HopToDesk

------------------
Hi Dan

At first, I didn't know that HoptoDesk was actualy a rebranded version of RustDesk. No problem, you have the right to do this. 
But you may have misunderstood the RustDesk license (which is AGPL version 3). Basicaly, you are not allowed to change copyright and you must make clear (in the source) that you modified a source code from RustDesk. AFAIK this is not the case.

I hope it was only a misunderstood. If you need help to understand the license and its requirements see https://www.gnu.org/licenses/licenses.html (AGPL3 is GPL3 with one addition).

I will not help you before you fix that.

Best regards

Le 22.09.2022 07:24, HopToDesk a écrit :
> Hi Xavier,
> 
> Thanks for reporting the issue. The typo is now fixed.  We have also
> updated the control file and included it on gitlab, and released a new
> .deb file with the changes. You may test again and let us know any
> issues.
> 
> 
> Yes, the source is in the source.zip file on gitlab, it would be great
> if you could help set it up properly, we could give you
> access/permissions, just let us know.
> 
> 
> Best Regards,
> 
> Dan B.
> HopToDesk
> 
> 
> ---- On  Tue, 20 Sep 2022 05:05:36 -0700
> "xavier"  wrote ----
> 
> 
>  >  Hello
>  > Here's some bug reports regarding debian package for hoptodesk.
>  > 
>  > - There is a typo in /usr/share/appliactions ; should be
> /usr/share/applications
>  > - missing dependencies on python3-pip and libxdo3
>  > 
>  > BTW, I've seen your repository on Github but no source files for
> development. Do you need help to set it up ?
>  > 
>  > Best Regards
>  > Xavier

------------------------------------

**Reply by zeroheure** [reply]
BTW the Github repo is a private copy of HopToDesk repo on Gitlab:
https://gitlab.com/hoptodesk/hoptodesk

**Reply by rustdesk** [reply]
Thanks, but i am curious, why did you use such an insane clone?

**Reply by zeroheure** [reply]
Well, I was curious about new RDS softwares (I often use DWService). My short list was alphabeticaly sorted, so HopTodesk was before RustDesk :-)

**Comment by zeroheure** [comment]
Just received this quick HoptoDesk answer:
>Thanks for pointing that out. We now updated the Cargo.toml file with the updated copyright and the notice in the About window to clarify the project is a fork of RustDesk. Hope it resolves the issue. From our understanding, the original copyright for RustDesk should remain, and we can add our additional copyright notice for our modifications to the project.

**Comment by lrq3000** [comment]
Thank you for doing that, and nice that they clarified the license. But it's sad that they are still trying to hide their lineage in all the (few) documentation they provide. If their fork really added a value, they would not fear to mention that they are a fork of RustDesk.

**Comment by dinger1986** [comment]
Any and all changes they make to any of the code should be made available as part of the AGPL, RustDesk can then choose to merge that or not

**Comment by lrq3000** [comment]
Yes of course, and it seems they are now complying with the license, I could verify that RustDesk is still mentioned in the About dialog. But so far it seems they only did a rebranding with a new mascot and icon.

**Reply by dinger1986** [reply]
Not really complying, they have been working on similar to RustDesk Server Pro and none of that's showing in the source. Any additions to the code should be added there

**Comment by kolAflash** [comment]
Related: #2778 _Warning about hoptodesk.com_
